### PR TITLE
oneAPI interfaces for Covariance

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            daal_covariance_kernel_t<Float>().compute(daal_data.get(),
+                                                      daal_cov_matrix.get(),
+                                                      daal_means.get(),
+                                                      &daal_parameter));
+        is_mean_computed = true;
+
+        result.set_cov_matrix(
+            homogen_table::wrap(arr_cov_matrix, component_count, component_count));
+    }
+    if (desc.get_result_options().test(result_options::cor_matrix)) {
```
Comment (PetrovKP) [inline cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp:77]: Why are you copied much code when the different is only in the parameter value `daal_parameter.outputMatrixType`?

File: cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            daal_covariance_kernel_t<Float>().compute(daal_data.get(),
+                                                      daal_cor_matrix.get(),
+                                                      daal_means.get(),
+                                                      &daal_parameter));
+        is_mean_computed = true;
+
+        result.set_cor_matrix(
+            homogen_table::wrap(arr_cor_matrix, component_count, component_count));
+    }
+    if (desc.get_result_options().test(result_options::means)) {
```
Comment (PetrovKP) [inline cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp:94]: Same comment

File: cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                                                      component_count * component_count,
+                                                      sycl::usm::alloc::device);
+
+            const auto daal_cov_matrix = interop::convert_to_daal_table(queue,
+                                                                        arr_cov_matrix,
+                                                                        component_count,
+                                                                        component_count);
+
+            interop::status_to_exception(
+                daal_covariance_kernel_t<Float>().compute(daal_data.get(),
```
Comment (PetrovKP) [inline cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp:106]: is the covariance always calculated? then why need this condition?

**Comment by PetrovKP** [comment]
Where dpcpp examples?

**Review by PetrovKP — APPROVED** [review]
please align the spaces in the covariance examples with other examples. very striking. and you can merge

**Comment by makart19** [comment]
please add your algo into cpp/oneapi/dal.hpp

**Pull request merged**